src/plugins/vis_augmenter/common/types.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ExpressionFunctionDefinition } from '../../expressions';
+
+export interface VisLayer {
+  // will be used as the column ID
+  id: string;
+  // will be used as the name when hovering over the tooltip
+  name: string;
+}
+
+export type VisLayers = VisLayer[];
+
+// resourceId & resourceName are required so that the
+// events flyout can partition data based on these attributes
+// (e.g., partitioning anomalies based on the detector they came from)
+export interface PointInTimeEventMetadata {
```

Comment:
nit - `EventMetadata` may be sufficiently clear.

Reply:
Right, I think I agree, keeping as `EventMetadata` could make it extendable if other event types have a similar structure as well